### DISCUSSION TITLE
How to style active handles differently

### DISCUSSION THREAD

**Discussion by what3v3rr** [xyflow/xyflow#2897] [Q&A]
Hi - I'm hoping to style "active" handles differently in the following scenarios:

- When the user is dragging from this handle
- When there is a connection between this handle and another handle

Here is an image to illustrate what I mean:
<img width="544" alt="Screenshot 2023-03-06 at 2 16 52 PM" src="https://user-images.githubusercontent.com/47459792/223253269-27c3a49e-67c1-42fe-ba70-fbd8e44eefec.png">

I looked at the [Validation examples here](https://reactflow.dev/docs/examples/interaction/validation/), but can't figure out how to adapt it for my use case. Any pointers would be appreciated!

**Comment by moklick** [comment]
A handle doesn't know if it's connected. You would need to check that on your own inside your custom nodes like:

```js
const isSourceConnected = useStore(s => s.edges.some(edge => e.source === id));
```

Does that help? I will check if we could add a `connected` class.

**Reply by agauravdev** [reply]
does this have performance impact? Lets say I have 20 nodes on my screen.. each node has a few handles.. whenever I change the node position, this function runs again, and it runs for all 4-5 handles in one node and continously as I drag my node.
